serving/src/test/java/feast/serving/it/ServingServiceIT.java

```diff
@@ -0,0 +1,555 @@
+              featureValue = ValueProto.Value.parseFrom(redisValueV);
+            }
+          } catch (InvalidProtocolBufferException e) {
+            e.printStackTrace();
+          }
+        }
+      }
+      // Check for null featureReference i.e key is not found
+      if (featureReference != null) {
+        Feature feature =
+            Feature.builder()
+                .setFeatureReference(featureReference)
+                .setFeatureValue(featureValue)
+                .setEventTimestamp(eventTimestamp)
+                .build();
+        retrievedFeatures.add(Optional.of(feature));
+      }
+    }
+
+    List<ValueProto.Value> expectedValues =
+        new ArrayList<>() {
+          {
+            add(feature1Value);
+            add(feature2Value);
+            add(feature3Value);
+          }
+        };
+    List<ServingAPIProto.FeatureReferenceV2> expectedFeatureReferences =
+        new ArrayList<>() {
+          {
+            add(feature1Reference);
+            add(feature2Reference);
+            add(feature3Reference);
+          }
+        };
+    assertEquals(3, retrievedFeatures.size());
+    assertEquals(
+        expectedFeatureReferences.stream()
+            .map(featureReference -> featureReference.getName())
+            .sorted()
+            .collect(Collectors.toList()),
+        retrievedFeatures.stream()
+            .map(feature -> feature.get().getFeatureReference().getName())
+            .sorted()
+            .collect(Collectors.toList()));
+    assertEquals(
+        expectedFeatureReferences.stream()
+            .map(featureReference -> featureReference.getFeatureTable())
+            .collect(Collectors.toList()),
+        retrievedFeatures.stream()
+            .map(feature -> feature.get().getFeatureReference().getFeatureTable())
+            .collect(Collectors.toList()));
+  }
+
+  @Test
+  public void shouldRegisterAndGetOnlineFeatures() {
+    ServingServiceGrpc.ServingServiceBlockingStub servingStub =
+        TestUtils.getServingServiceStub(false, FEAST_SERVING_PORT, null);
+
+    // getOnlineFeatures Information
+    String projectName = "default";
+    String featureTableName = "rides";
+    String entityName = "driver_id";
+    ValueProto.Value entityValue = ValueProto.Value.newBuilder().setInt64Val(1).build();
+    String feature1 = "trip_cost";
+
+    // Instantiate EntityRows
+    GetOnlineFeaturesRequestV2.EntityRow entityRow1 =
+        GetOnlineFeaturesRequestV2.EntityRow.newBuilder()
+            .setTimestamp(Timestamp.newBuilder().setSeconds(100))
+            .putFields(entityName, entityValue)
+            .build();
+    List<GetOnlineFeaturesRequestV2.EntityRow> entityRows =
+        new ArrayList<>() {
+          {
+            add(entityRow1);
+          }
+        };
+
+    // Instantiate FeatureReferences
+    ValueProto.Value feature1Value = ValueProto.Value.newBuilder().setDoubleVal(42.2).build();
```

Comment:
Please use some helpers / create on in `common-test`.
Test code should be readable